pkg/actions/lua_test.go

```diff
@@ -88,6 +110,197 @@ func TestLuaRun(t *testing.T) {
+	})
+	err = h.Run(ctx, graveler.HookRecord{
+		RunID:            "abc123",
+		EventType:        graveler.EventTypePreCreateBranch,
+		RepositoryID:     "example123",
+		StorageNamespace: "local://foo/bar",
+		SourceRef:        "abc123",
+		BranchID:         "my-branch",
+		Commit: graveler.Commit{
+			Version: 1,
+		},
+		CommitID: "123456789",
+		PreRunID: "3498032432",
+		TagID:    "",
+	}, out)
+	const expectedErr = "module 'net/http' not found"
+	if err == nil || !strings.Contains(err.Error(), expectedErr) {
+		t.Fatalf("Error=%v, expected: '%s'", err, expectedErr)
+	}
+}
+
+func TestLuaRun_NetHttp(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		codeVal := r.URL.Query().Get("code")
+		statusCode, _ := strconv.Atoi(codeVal)
+		if statusCode != 0 {
+			w.WriteHeader(statusCode)
+		}
+		_, _ = fmt.Fprint(w, "hello-"+r.Method)
+		body, _ := io.ReadAll(r.Body)
+		if len(body) > 0 {
+			_, _ = w.Write([]byte{' '})
+			_, _ = w.Write(body)
+		}
+	}))
+	defer ts.Close()
+
+	tests := []struct {
+		Name     string
+		Script   string
+		Expected string
+	}{
+		{
+			Name: "simple_get",
+			Script: `local http = require("net/http")
+local code, body, headers, status = http.request("` + ts.URL + `")
+print(code .. " " .. body .. " " .. status)
+`,
+			Expected: "200 hello-GET 200 OK",
+		},
+		{
+			Name: "simple_post",
+			Script: `local http = require("net/http")
+local code, body, headers, status = http.request("` + ts.URL + `", "name=value")
+print(code .. " " .. body .. " " .. status)
+`,
+			Expected: "200 hello-POST name=value 200 OK",
+		},
+		{
+			Name: "simple_get_404",
+			Script: `local http = require("net/http")
+local code, body, headers, status = http.request("` + ts.URL + `/?code=404")
+print(code .. " " .. body .. " " .. status)
+`,
+			Expected: "404 hello-GET 404 Not Found",
+		},
+		{
+			Name: "table_get",
+			Script: `local http = require("net/http")
+local code, body, headers, status = http.request{
+	url="` + ts.URL + `",
+	method="GET",
+}
+print(code .. " " .. body .. " " .. status)
+`,
+			Expected: "200 hello-GET 200 OK",
+		},
+		{
+			Name: "table_post",
+			Script: `local http = require("net/http")
+local code, body, headers, status = http.request{
+	url="` + ts.URL + `",
+	body="name=value",
+}
+print(code .. " " .. body .. " " .. status)
+`,
+			Expected: "200 hello-POST name=value 200 OK",
+		},
+		{
+			Name: "table_post_method",
+			Script: `local http = require("net/http")
+local code, body, headers, status = http.request{
+	url="` + ts.URL + `",
+	method="POST",
+	body="name=value",
+}
+print(code .. " " .. body .. " " .. status)
+`,
+			Expected: "200 hello-POST name=value 200 OK",
+		},
+		{
+			Name: "table_get_404",
+			Script: `local http = require("net/http")
+local code, body, headers, status = http.request{
+	url="` + ts.URL + `/?code=404",
+}
+print(code .. " " .. body .. " " .. status)
+`,
+			Expected: "404 hello-GET 404 Not Found",
+		},
```

Comment:
Maybe add a 500 reponse

Reply:
It will be equal to the 404 test - but I think I got what you wanted - request that will produce an error not just non-success http response.